app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_dict = {}
+
+    def inner(*args, **kwargs) -> Any:
+        if args not in cache_dict:
+            cache_dict[args] = func(*args)
+            print("Calculating new result")
+            return cache_dict[args]
+        else:
+            print("Getting from cache")
+            return cache_dict[args]
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.